src/Bicep.LangServer/Handlers/BicepHoverHandler.cs

```diff
@@ -112,6 +107,28 @@ public BicepHoverHandler(ISymbolResolver symbolResolver)
             }
         }
 
+        
+        //we need to check for overflow due to using code blocks.
+        //if we reach limit in a code block vscode will truncate it automatically, the block will not be terminated so the hover will not be properly formatted
+        //therefore we need to check for the limit ourselves and truncate text inside code block, making sure it's terminated properly.
+        private static string CodeBlock(string content) =>
+        $"```bicep\n{(content.Length > MaxHoverMarkdownCodeBlockLength ? content.Substring(0, MaxHoverMarkdownCodeBlockLength) : content)}\n```";
```

Comment:
I think when emitting markdown, it's best to stick to either "each block includes a trailing newline", or "each block includes a preceding newline" - makes things more predictable when joining blocks together, so there are fewer edge cases.

Here it looks like the code originally used a trailing newline, so I think it would be best to stick with that approach.